### PR TITLE
Store FromCollectiveId/HostCollectiveId/OrderId for Activities

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/5808
Resolve https://github.com/opencollective/opencollective/issues/5822

**Todo**
- [x] Should we add another `MemberCollectiveId` column? => keeping it simple for now, the migration will be easy if needed
- [x] Try mutation on local prod DB copy
- [x] Review all places where we do `HostCollectiveId: collective.HostCollectiveId`, it's unsafe to do that without checking `isActive`
- [x] Expose `fromAccount`, `host`, `order`, `expense` on GQLV2's `Activity`
- [ ] Create follow-up issue to migrate past transaction activities

**Migration benchmark**

```
Creating columns: 9.282ms
Migrating User actions where FromCollectiveId/CollectiveId should be user's profile: 1.489s
Migrating Transaction activities: 10:46.391 (m:ss.mmm) **EDIT: 24s after latest filtering**
Migrating gift cards: 25.139ms
Migrating members: 32.421s
Migrating Virtual Cards / Freeze collective activities: 102.684ms
Migrating subscriptions & rejected contributions: 1.691s
Migrating expense activities: 33.269s
Migrating expense moved root activities: 0.679ms
Migrating Orders activities from data: 757.843ms
Migrating Orders activities from data: 757.843ms
Migrating Orders activities from joins: 26.327s
Migrating contact activities: 732.448ms
Migrating comment/conversation activities: 1.399s
Migrating updates activities: 3.213s
Migrating host actions activities: 3.614s
Migrating host status activities: 1.262s
Populating HostCollectiveId for activities using JOIN: 54.759s
```

### CODE REVIEW

File: migrations/20220816142946-enhance-activities-links.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+      UPDATE "Activities" a
+      SET
+        "OrderId" = t."OrderId",
+        "CollectiveId" = t."CollectiveId",
+        "FromCollectiveId" = t."FromCollectiveId",
+        "HostCollectiveId" = t."HostCollectiveId"
+      FROM "Transactions" t
+      WHERE a."TransactionId" IS NOT NULL
+      AND t.id = a."TransactionId"
+      AND a."createdAt" >= '2022-06-01' -- This migration is too heavy, we'll run the rest separately
```
Comment (Betree) [inline migrations/20220816142946-enhance-activities-links.js:63]: I'll run the rest of this one progressively/manually

File: migrations/20220816142946-enhance-activities-links.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+      type: Sequelize.INTEGER,
+      references: { key: 'id', model: 'Collectives' },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: true,
+    });
+
+    console.timeEnd('Creating columns');
+
+    // -- Fill in columns from existing data --
```
Comment (SudharakaP) [inline migrations/20220816142946-enhance-activities-links.js:34]: @Betree : Maybe we could also do this same fill in columns from existing data for `collective.created` activity type. Note that the `CollectiveId` is contained within the `data` field there.

Comment (Betree) [reply migrations/20220816142946-enhance-activities-links.js:34]: `collective.created` is covered in https://github.com/opencollective/opencollective-api/blob/53cda4a23a1b35c708d5ee18b9a7218b03eaf578/migrations/20220816142946-enhance-activities-links.js#L285

Comment (SudharakaP) [reply migrations/20220816142946-enhance-activities-links.js:34]: My thinking is that, `data.host.id` only exists for `collective.apply` activity. It seems that for `collective.created` what we need to copy is `data.collective.id` (to `CollectiveId` column).

**Pull request merged**